astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -533,26 +519,24 @@ def test_teme_itrf():
 
 
 @pytest.mark.remote_data
-def test_earth_orientation_table():
+def test_earth_orientation_table(monkeypatch):
     """Check that we can set the IERS table used as Earth Reference.
 
     Use the here and now to be sure we get a difference.
     """
+    monkeypatch.setattr('astropy.utils.iers.conf.auto_download', True)
     t = Time.now()
     location = EarthLocation(lat=0*u.deg, lon=0*u.deg)
     altaz = AltAz(location=location, obstime=t)
     sc = SkyCoord(1*u.deg, 2*u.deg)
     # Default: uses IERS_Auto, which will give a prediction.
-    with catch_warnings() as w:
-        altaz_auto = sc.transform_to(altaz)
-
-    assert len(w) == 0
+    # Note: tests run with warnings turned into errors, so it is
+    # meaningful if this passes.
+    altaz_auto = sc.transform_to(altaz)
 
     with iers.earth_orientation_table.set(iers.IERS_B.open()):
-        with catch_warnings() as w:
+        with pytest.warns(AstropyWarning, match='after IERS data'):
             altaz_b = sc.transform_to(altaz)
-        assert len(w) == 1
```

Comment:
So you don't want to check to see if this is the *only* warning emitted anymore?

Reply:
I think it is fine to just test that we get the right warning. If there are others, pytest would presumably fail.

Reply:
Not really. Consider this example:

```python
import warnings

import pytest


def myfunc():
    warnings.warn('Catch this', UserWarning)
    warnings.warn('Nope', UserWarning)  # Even if you change the warning type, same result below


def test1():
    with pytest.warns(UserWarning, match='Catch this'):
        myfunc()  # passes


def test2():
    with pytest.warns(UserWarning, match='Catch this') as w:
        myfunc()
    assert len(w) == 1  # fails
```

Reply:
It comes down to whether you care if the functionality emits other warnings you are not checking for or not.

Reply:
I think in this case we check the code enough in other places that I don't worry, but clearly that is not the behaviour I had expected....